### PR TITLE
Public type definitions

### PR DESCRIPTION
**Pull request by teogeb**
Define valid types for public API methods

Also small bug fix for parsing stream definition options (e.g. when filtering subscriptions with `getSubscription(opts)`)

### CODE REVIEW

File: src/StreamrClient.ts [hunk 1]
```diff
@@ -18,21 +18,29 @@ import { DataUnion, DataUnionDeployOptions } from './dataunion/DataUnion'
...
 // TODO get metadata type from streamr-protocol-js project (it doesn't export the type definitions yet)
 export type OnMessageCallback = MaybeAsync<(message: any, metadata: any) => void>
 
+// TODO check if these are correct
+export interface SubscribeOptions {
+    resend?: boolean
+    from?: { timestamp: number, sequenceNumber?: number }
+    to?: { timestamp: number, sequenceNumber?: number }
+    last?: number
+}
```
Comment (timoxley) [inline src/StreamrClient.ts:32]: `resend` is not a boolean.

basically, resend options can either be nested or top-level in the config.

```typescript
type ResendOptions = {
    from?: { timestamp: number, sequenceNumber?: number }
    to?: { timestamp: number, sequenceNumber?: number }
    last?: number
}

type SubscribeOptions = {
    resend?: ResendOptions
} & ResendOptions
```

File: test/unit/StreamUtils.test.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { Stream } from '../../src/stream'
+import { validateOptions } from '../../src/stream/utils'
+
+describe('Stream utils', () => {
```
Comment (timoxley) [inline test/unit/StreamUtils.test.ts:4]: Good

**Review by timoxley — CHANGES_REQUESTED** [review]
resend options type is not right, but otherwise lgtm

File: src/subscribe/index.ts [hunk 3]
```diff
@@ -126,6 +126,7 @@ export class Subscription extends Emitter {
         return this.pipeline.throw(...args)
     }
 
+    // TODO should we expose this to the user as no-args method?
```
Comment (teogeb) [inline src/subscribe/index.ts:129]: @timoxley How about this `subscription.unsubscribe()`? Is there some cases where user would this with parameters? At least internally we can call this with an `error` parameter, if I understood correctly. (Does that error parameter have some specific type?)

File: src/StreamrClient.ts (hunk 1 above)
Comment (teogeb) [reply src/StreamrClient.ts:30]: Fixed

**Review by timoxley — APPROVED** [review]
👍 lgtm, good stuff.

**Pull request merged**